src/loops.rs

```diff
@@ -161,8 +162,21 @@ impl LateLintPass for LoopsPass {
         // check for `loop { if let {} else break }` that could be `while let`
         // (also matches explicit "match" instead of "if let")
         if let ExprLoop(ref block, _) = expr.node {
-            // extract a single expression
-            if let Some(inner) = extract_single_expr(block) {
+            // extract a single expression (or statement)
+            let inner_stmt = extract_single_expr_from_stmt(block);
+            let inner_expr = extract_single_expr(block);
+            let extracted = match inner_stmt {
+                Some(_) => inner_stmt,
+                None => inner_expr,
+            };
+
+            if let Some(inner) = extracted {
+                // collect remaining expressions below the match
+                let other_stuff = (1..block.stmts.len())
+                                  .map(|i| {
```

Comment:
You're in good company. When we started using clippy on servo and rustc, we got a lot of warnings, many of them legit.